libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/http/MockWebServerRule.kt

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.navigation.testing.ui.http
+
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+/**
+ * Creates and initializes a [MockWebServer] for each test.
+ *
+ * All request that should be mocked need to have a base url equal to [baseUrl]
+ * and have a [MockRequestHandler] that can serve the request added to the [requestHandlers] list.
+ */
+class MockWebServerRule : TestWatcher() {
+
+    val webServer = MockWebServer()
```

Comment:
> Property 'webServer' could be private

Reply:
Leaving the public to indicate that it can be manipulated as the test needs.